Utilities/src/main/java/edu/stanford/hivdb/mutations/Mutation.java

```diff
@@ -103,26 +103,21 @@ public Mutation(Gene gene, int pos, Character aa) {
 		this(gene, pos, "" + aa, "", "");
 	}
 	
-	// Potential Issue:
-	// Doesn't support sole insertions such as "_N"
-	// See commented-out conditional for illustration
 	public static Mutation fromNucAminoMutation(Gene gene, int aaStart, Map<?, ?> mut) {
 		int pos = ((Double) mut.get("Position")).intValue() - aaStart + 1;
-		
 		String codon = "";
 		String insertedCodon = "";
 		boolean isInsertion = (Boolean) mut.get("IsInsertion");
 		boolean isDeletion = (Boolean) mut.get("IsDeletion");
-		
 		StringBuilder aas = new StringBuilder();
-		if (isDeletion) {
-			aas.append('-');
-		}
+		if (isDeletion) aas.append('-');
 		else {
-			codon = (String) mut.get("CodonText");
-			codon = codon.replace(' ', '-');
-			/*if (codon.length() > 0)*/ aas.append(CodonTranslation.translateNATriplet(codon));
-			if (isInsertion) {       
+			if (mut.containsKey("CodonText")) {
```

Comment:
I revisited my tests for this method and took a deeper look last night. The commented-out conditional we discussed yesterday was meant to allow the method to called without needing to specify a codon if, for example, the mutation is simply a single insertion such as _N-- is that a valid case?

Reply:
Insertion means a mutation (which already has one codon) with extra codon(s), therefore "_N--" is not a valid case. NucAmino guaranteed that a non-deletion mutation always has a `CodonText` consisted by three nucleic acids ([nucamino/.../mutation.go](https://github.com/hivdb/nucamino/blob/b18ebbb6fb179ef89906c185964e4175654d78d4/types/mutation/mutation.go#L95), see L105 and L136). For a deletion, the `CodonText` exists but is consistently an empty string ([nucamino/.../mutation.go#L86](https://github.com/hivdb/nucamino/blob/b18ebbb6fb179ef89906c185964e4175654d78d4/types/mutation/mutation.go#L86)).

In general, `CodonText` is always supplied by NucAmino, no matter insertion, deletion or plain mutation.

Reply:
Ah this makes perfect sense. I read through some of the NucAmino code last week and saw that codons always consist of three nas (L108). Though I didn't notice the default empty string case. Thank you for clarifying! I've reverted the implementation back and removed the comment. (Though I know we're refactoring anyways)

Reply:
This method still exists in `IUPACMutation`.

Reply:
Awesome. I haven't gotten a chance to review that class yet. Thanks for pointing that out :)